types/live-server/index.d.ts

```diff
@@ -1,28 +1,44 @@
+/// <reference types="node" />
+
+import * as http from "http";
+
+export class Request extends http.IncomingMessage {
+    originalUrl?: http.IncomingMessage["url"];
+}
+
+export class Response extends http.ServerResponse {}
+
+export type NextFunction = (err?: any) => void;
+
 /**
  * The live-server start params.
  */
-export interface LiveServerParams {
-    /** Set the server port. Defaults to 8080. */
-    port?: number | undefined;
-    /**  Set the address to bind to. Defaults to 0.0.0.0 or process.env.IP. */
-    host?: string | undefined;
-    /** Set root directory that's being served. Defaults to cwd. */
-    root?: string | undefined;
-    /** When false, it won't load your browser by default. */
-    open?: boolean | undefined;
-    /** Comma-separated string for paths to ignore. */
-    ignore?: string | undefined;
-    /** When set, serve this file (server root relative) for every 404 (useful for single-page applications). */
-    file?: string | undefined;
-    /** Waits for all changes, before reloading. Defaults to 0 sec. */
-    wait?: number | undefined;
-    /** Mount a directory to a route. */
-    mount?: string[][] | undefined;
-    /** 0 = errors only, 1 = some, 2 = lots */
-    logLevel?: 0 | 1 | 2 | undefined;
-    /** Takes an array of Connect-compatible middleware that are injected into the server middleware stack. */
-    middleware?: Array<(req: any, res: any, next: any) => void> | undefined;
-}
+export interface LiveServerParams extends
```

Comment:
I feel like the original way of writing this is clearer? It also means that it's possible to add certain required fields, if that's ever desired

Reply:
I just feel that the shape it has now is cleaner. Anyway, leaving aside "what we feel", the truth is that by using `Partial` we are leaving the code cleaner, less vebose and simpler.

On the other hand, regarding if you ever want to add mandatory fields, you coild do the following:
```ts
type ParamsRequired = {
    // ...
}

type ParamsPartial = Partial<{
    port: number;
    host: string;
    // ...
}>

export interface LiveServerParams extends ParamsRequired, ParamsPartial { }
```
But I didn't do it because I didn't see it necessary, at least not at this moment.

Reply:
This code style is highly unconventional, it is objectively _not_ simpler or more concise, and all of the people who control the fate of this PR disagree with you that it’s cleaner. I’m sorry you disagree with this decision, but when you make a contribution to a codebase you don’t own, you have to be willing to conform to the existing style and the wishes of the maintainers. You can change it if you want to continue working on this PR; otherwise the bot will close it soon.